Firestore/Source/Remote/FSTRemoteEvent.mm

```diff
@@ -605,44 +541,91 @@ - (void)recordResponseForTargetID:(FSTBoxedTargetID *)targetID {
  * preventing in preventing race conditions for a target where events arrive but the server hasn't
  * yet acknowledged the intended change in state.
  */
-- (BOOL)isActiveTarget:(FSTBoxedTargetID *)targetID {
+- (BOOL)isActiveTarget:(FSTTargetID)targetID {
   return [self queryDataForActiveTarget:targetID] != nil;
 }
 
-- (FSTQueryData *_Nullable)queryDataForActiveTarget:(FSTBoxedTargetID *)targetID {
-  FSTQueryData *queryData = self.listenTargets[targetID];
-  return (queryData && !self.pendingTargetResponses[targetID]) ? queryData : nil;
+- (FSTQueryData *_Nullable)queryDataForActiveTarget:(FSTTargetID)targetID {
+  const auto &targetState = _targetStates.find(targetID);
```

Comment:
Again `auto targetState`.

In general, make the types of the locals match the types of the return type.